### PR TITLE
Add maintenance page feature flag

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Reroute all requests to maintenance page based on flag

#### Ticket

[CBO-996](https://dsdmoj.atlassian.net/browse/CBO-996)

#### Why
So we can control access to the TD production
and Kuberentes/live-1 production sites using the
app itself.

Previous incarnation involved changing DNS A records
to divert all traffic to a static web site hosted
in s3, using an AWS CloudFront distribution. This
method is onerous to setup and the migraiton to live-1
would require two such setups.

#### How
see Maintenance mode section of readme

copied here for conveninence:

## Maintenance mode

A Conditional catchall routes exists in `routes.rb`. This directs all routes requested to the `pages#servicedown` controller and view. To activate the conditional route you must provide the app server with either MAINTENANCE_MODE=true (kubernetes) or DOCKER_STATE=maintenance (template-deploy). Note that dotenv files cannot be used to set these envvars locally as the config gem (`settings.yml` file) is loaded before dotenv files.

```bash
# activate maintenance mode locally
MAINTENANCE_MODE=true rails s
```

You can deploy the app in maintenance mode for both template-deploy orchestrated environments and kubernetes orchestrated environments:

### Template-deploy maintenance mode
- build the app using Jenkins as normal.
- deploy the app to an environment selecting `maintenance` for the task option (jenkins string parameter)

Note that this is a quick fix method that leverages templates-deploy pre-existing environment variable `DOCKER_STATE` for purposes it was not intended for.

### kubernetes maintenance mode
You can switch on maintenance mode by deploying either from your local machine or via circleCI. Either
method requires you to amend the `deployment.yaml` file for the relevant environment, adding `MAINTENANCE_MODE=true`. e.g. `./kubernetes_deploy/dev/deployment.yaml`

To apply via circleCI:

- commit the above change and push to github
- run through circleCI and deploy to the relevant environment
- For production environment you will need to merge to master and use its workflow

To apply via local machine:

- you will need all relevant aws credentials and git-crypted secrets access
- use the deploy script

```bash
 # put dev into maintenance mode
 kubernetes_deploy/scripts/deploy.sh dev latest
```

### CODE REVIEW

File: spec/requests/pages/servicedown_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    context '/api/case_types' do
+      before { get '/api/case_types', params: { api_key: user.persona.provider.api_key, format: :json } }
+
+      it { expect(response).to have_http_status :service_unavailable }
+      it { expect(response.body).to be_json }
+      it { expect(response.body).to include_json({error: "Temporarily unavailable"}.to_json) }
+    end
+  end
+end
```
Comment (jsugarman) [inline spec/requests/pages/servicedown_spec.rb]: linebreak

File: spec/requests/pages/servicedown_spec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+      let(:user) { create(:case_worker).user }
+        context '/case_workers/claims' do
+        before { get case_workers_home_path }
+        it_behaves_like 'maintenance page'
+      end
+    end
+
+    context 'advocate' do
+      before { sign_in user }
```
Comment (jsugarman) [inline spec/requests/pages/servicedown_spec.rb:59]: not needed?

File: spec/requests/pages/servicedown_spec.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+      context '/advocates/claims/new' do
+        before { get new_advocates_claim_path }
+        it_behaves_like 'maintenance page'
+      end
+    end
+  end
+
+  context 'api requests (json)' do
+    before { sign_in user }
```
Comment (jsugarman) [inline spec/requests/pages/servicedown_spec.rb]: not needed for api call?

**Review by naseberry — APPROVED** [review]
:+1: LGTM

**Pull request merged**